component/client/settings/ClientSetting.tsx

```diff
@@ -93,7 +94,8 @@ class ClientSetting extends React.Component<any, IClientSettingState> {
     event.preventDefault();
     if (event.target.value !== undefined) {
       this.setState({
-        dirty: this.state.username.original !== event.target.value,
+        // The field is only considered dirty if there is value and it's different from original
+        dirty: event.target.value && this.state.username.original !== event.target.value,
         username: {
           original: this.state.username.original,
```

Comment:
Agreed. This is why you frequently see code in which they extract pieces of the state into a `const`.